packages/@sanity/form-builder/src/inputs/NumberInput.tsx

```diff
@@ -1,56 +1,53 @@
 import React from 'react'
-import {uniqueId} from 'lodash'
 import {NumberSchemaType} from '@sanity/types'
-import TextInput from 'part:@sanity/components/textinputs/default'
-import FormField from 'part:@sanity/components/formfields/default'
+import {TextInput} from '@sanity/ui'
+import {useId} from '@reach/auto-id'
+import {FormField} from '@sanity/base/components'
 import {getValidationRule} from '../utils/getValidationRule'
 import PatchEvent, {set, unset} from '../PatchEvent'
 import {Props} from './types'
 
-export default class NumberInput extends React.Component<Props<number, NumberSchemaType>> {
-  _input: TextInput | null
-  _inputId = uniqueId('NumberInput')
-  handleChange = (event: React.SyntheticEvent<HTMLInputElement>) => {
-    const nextValue = event.currentTarget.value
-    this.props.onChange(PatchEvent.from(nextValue === '' ? unset() : set(Number(nextValue))))
-  }
-  focus() {
-    if (this._input) {
-      this._input.focus()
-    }
-  }
-  setInput = (input: TextInput | null) => {
-    this._input = input
-  }
-  render() {
-    const {value = '', readOnly, markers, type, level, onFocus, presence} = this.props
-    const validation = markers.filter((marker) => marker.type === 'validation')
-    const errors = validation.filter((marker) => marker.level === 'error')
-    // Show numpad on mobile if only positive numbers is preferred
-    const minRule = getValidationRule(type, 'min')
-    const onlyPositiveNumber = minRule && minRule.constraint >= 0
-    return (
-      <FormField
-        markers={markers}
-        level={level}
-        label={type.title}
-        description={type.description}
-        labelFor={this._inputId}
-        presence={presence}
-      >
-        <TextInput
-          inputId={this._inputId}
-          customValidity={errors && errors.length > 0 ? errors[0].item.message : ''}
-          type="number"
-          value={value}
-          readOnly={readOnly}
-          placeholder={type.placeholder}
-          onChange={this.handleChange}
-          onFocus={onFocus}
-          ref={this.setInput}
-          pattern={onlyPositiveNumber ? '[d]*' : undefined}
-        />
-      </FormField>
-    )
-  }
-}
+const NumberInput = React.forwardRef(function NumberInput(
+  props: Props<number, NumberSchemaType>,
+  forwardedRef: React.ForwardedRef<HTMLInputElement>
+) {
+  const {value = '', readOnly, markers, type, level, onFocus, onChange, presence} = props
+  const validation = markers.filter((marker) => marker.type === 'validation')
+  const errors = validation.filter((marker) => marker.level === 'error')
+  const id = useId()
+  // Show numpad on mobile if only positive numbers is preferred
```

Comment:
Is this comment still accurate?

Reply:
Looks like it? It was introduced in db3d22875ce9f8d2fdb983ed8a741ec84fa86cb3 and I'm thinking it's out of scope to change this as part of this PR?

Reply:
Ah, you mean after adding `inputMode="numeric"`. Good point. I'm changing to `text` in the case of `onlyPositiveNumber`